themes/V3/5ePHB/widgets.js

```diff
@@ -0,0 +1,24 @@
+const { WIDGET_TYPE, FIELD_TYPE } = require('../../../shared/naturalcrit/codeEditor/helpers/widget-elements/constants');
+
+module.exports = [{
+	name    : 'monster',
+	type    : WIDGET_TYPE.SNIPPET,
+	classes : ['frame', 'wide']
```

Comment:
Alright, so after going through the video, I have some concern about the extensibility of this approach dividing the widgets up to have a `classes` property (that makes checkboxes) and a separate `fields` property (that makes textboxes). I foresee eventually running into confusion when we add other types of entry methods (is a colorpicker a "field"? is a watercolor image selector a "class"? Are there cases where we want checkboxes for fields and not just classes?

I propose reorganizing these properties into a single `widgets` array where each component is identified by its button type (checkbox/textbox), rather than its CSS type (class/field):
```suggestion
	name    : 'monster',
	type    : WIDGET_TYPE.SNIPPET,
	widgets : [
		{
			name : frame,
			type : checkbox
		},
		{
			name : wide,
			type : checkbox
		},
		{
			name : opacity,
			type : textbox
		},
	]
```

I think this could greatly simplify the `createWidgets()` step

Reply:
So in `index.js` where you define how to render the `cClass` element and the `field` element we would rename those to `checkbox` and `textbox`, and then later add `colorselector` and `imageselector`, etc.

Reply:
Yep that all makes sense. My one comment on it all is that I'd prefer we stick to the actual HTML input types -- so `checkbox` is fine, but I'd suggest `text` instead of `textbox`. And then for `imageselector` we'd use `select` and have appropriate additional fields on the object for options with name/value/image/etc...

Reply:
My only comment to that is we might run into some custom input types (like maybe "color selector" is one of those) that doesn't have a standard HTML Element. But deciding on a name for that can be handled when we get to it.